JenkinsFile

```diff
@@ -0,0 +1,203 @@
+                cf push ${fullAppName} -f manifest.jenkins.yml --hostname ${fullAppName} -d ${pcfDomain} --no-start
+                cf set-env ${fullAppName} SPACE ${pcfSpace}
+                cf set-env ${fullAppName} DOMAIN ${pcfDomain}
+                cf start ${fullAppName}
+              """
+          }
+      }
+      withCredentials([
+        [$class: 'StringBinding', credentialsId: "${env.THREADFIX_API_KEY}", variable: "THREADFIX_KEY"]
+      ]) {
+          sh """
+            mkdir -p ${root}/zap-out
+            ${zapHome}/zap.sh -cmd -quickout ${root}/zap-out/zap.xml -quickurl https://${fullAppName}.${pcfDomain}
+            cat ${root}/zap-out/zap.xml
+            /bin/curl -v --insecure -H 'Accept: application/json' -X POST --form file=@${root}/zap-out/zap.xml ${env.THREADFIX_URL}/rest/applications/${THREADFIX_ID}/upload?apiKey=${THREADFIX_KEY}
+          """
+      }
+      withEnv([
+        "CF_HOME=.cf"
+      ]) {
+          withCredentials([
+            [$class: 'UsernamePasswordMultiBinding', credentialsId: "${env.PCF_CREDS}", usernameVariable: "CFUSER", passwordVariable: "CFPASS"]
+          ]) {
+              sh """
+                cf api ${env.PCF_API_ENDPOINT}
+                cf auth ${CFUSER} ${CFPASS}
+                cf target -o ${env.PCF_ORG} -s ${pcfSpace}
+              """
+              def legacyAppNames = sh(script: """cf routes | grep \"${appName} \" | awk '{print \$4}'""", returnStdout: true)
+              sh "cf map-route ${fullAppName} ${pcfDomain} --hostname bf-ia-broker"
+              // Remove legacy applications
+              for (Object legacyApp : legacyAppNames.trim().tokenize(',')) {
+                  def legacyAppName = legacyApp.toString().trim()
+                  if(legacyAppName != appName) {
+                      sh "cf unmap-route ${legacyAppName} ${pcfDomain} --hostname ${legacyAppName}"
+                      sh "cf delete -f ${legacyAppName}"
+                  }
+              }
+          }
+      }
+    }
+
+    def integrationTestPhase = { String pcfSpace, String pcfDomain ->
+      withCredentials([
+        file(credentialsId: "${POSTMAN_SECRET_FILE}", variable: "POSTMAN_FILE")
+      ]) {
+          withEnv([
+            "PATH+=${root}/integration-tests/node_modules/newman/bin:${nodejs}/bin",
+            "PCF_SPACE=${pcfSpace}",
+            "HOME=${WORKSPACE}"
+          ]) {
+              dir ("integration-tests") {
+                  sh "ci/blackbox.sh"
+              }
+          }
+      }
+    }
+
+    stage("Phase One Deploy") {
+      deployPhase(env.PHASE_ONE_PCF_SPACE, env.PHASE_ONE_PCF_DOMAIN)
+    }
+    if(!params.SKIP_INTEGRATION_TESTS) {
+        stage("Phase One Integration Tests") {
+          integrationTestPhase(env.PHASE_ONE_PCF_SPACE, env.PHASE_ONE_PCF_DOMAIN)
+        }
+    }
+    stage("Phase Two Deploy") {
+      deployPhase(env.PHASE_TWO_PCF_SPACE, env.PHASE_TWO_PCF_DOMAIN)
+    }
+    if(!params.SKIP_INTEGRATION_TESTS) {
```

Comment:
Also use the same check https://github.com/venicegeo/pz-gateway/blob/master/JenkinsFile#L168 before doing the phase 2 integration tests